src/strategies/StrategyALL.java

```diff
@@ -0,0 +1,32 @@
+package strategies;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class StrategyALL implements SearchStrategy {
+
+    @Override
+    public void search(ArrayList<String> list, Map<String, ArrayList<Integer>> searchMap, String query) {
+        String[] words = query.split(" +");
```

Comment:
There is a special character for the `white space` - `\s`, please use it.

Reply:
Also, the line may start with white space, so think about it.

Reply:
This regex is used in different places. You can make it constant or use it in one place.

Reply:
fixed

Reply:
First and second comments are not fixed, left comments in other places, so close the conversation here.